concent_api/api-integration-test.py

```diff
@@ -0,0 +1,208 @@
+CONCENT_PRIVATE_KEY = b'[\x10!\xca\rv\xb1\x8d7\x84\xd2\x83\xb1\xa5\xab\xf2\x1c\xe0\xb0\x04Pj>hE|tr~,^\xb5'
+
+(PROVIDER_PRIVATE_KEY,  PROVIDER_PUBLIC_KEY)  = generate_ecc_key_pair()
+(REQUESTOR_PRIVATE_KEY, REQUESTOR_PUBLIC_KEY) = generate_ecc_key_pair()
+
+
+def print_golem_message(message, indent = 4):
+    assert isinstance(message, Message)
+
+    HEADER_FIELDS  = ['timestamp', 'encrypted', 'sig']
+    PRIVATE_FIELDS = {'_payload', '_raw', 'result_hash'}
+    assert 'type' not in message.__slots__
+    fields = ['type'] + HEADER_FIELDS + sorted(set(message.__slots__) - set(HEADER_FIELDS) - PRIVATE_FIELDS)
+    values = [
+        type(message).__name__ if field == 'type' else
+        getattr(message, field)
+        for field in fields
+    ]
+
+    for field, value in zip(fields, values):
+        if isinstance(value, bytes):
+            try:
+                nested_message = load(
+                    value,
+                    CONCENT_PRIVATE_KEY,
+                    CONCENT_PUBLIC_KEY,
+                )
+            except AttributeError:
+                # FIXME: golem-messages provides no reliable way to discern invalid messages from other AttributeErrors.
+                print('{}{:30} = <BINARY DATA>'.format(' ' * indent, field))
+            else:
+                print('{}{:30} ='.format(' ' * indent, field))
+                print_golem_message(nested_message, indent = indent + 4)
+        else:
+            print('{}{:30} = {}'.format(' ' * indent, field, value))
+
+
+def api_request(host, endpoint, private_key = None, data = None, headers = None, ):
+    assert all(value not in ['', None] for value in [endpoint, host, headers])
+    import ipdb; ipdb.set_trace()
+    url = "{}/api/v1/{}/".format(host, endpoint)
+
+    if data is None:
+        print('RECEIVE ({})'.format(url))
+    else:
+        print('SEND ({})'.format(url))
+        print('MESSAGE:')
+        print_golem_message(data)
+
+        data = dump(
+            data,
+            private_key,
+            CONCENT_PUBLIC_KEY,
+        )
+
+    if data is None:
+        response = requests.post("{}".format(url), headers = headers)
+    else:
+        response = requests.post("{}".format(url), headers = headers, data = data)
+
+    if len(response.content) != 0:
+        decoded_response = load(
+            response.content,
+            CONCENT_PRIVATE_KEY,
+            CONCENT_PUBLIC_KEY,
+        )
+        print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
+        print('MESSAGE:')
+        print_golem_message(decoded_response)
+    else:
+        print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
+        if response.text not in ['', None]:
+            print('RAW RESPONSE: {}'.format(response.text))
+    print()
+
+
+def parse_command_line(command_line):
```

Comment:
@cameel argparse.ArgumentParser() should do the job, but maybe you guys agreed to implement this in more custom way ?